core/views.py

```diff
@@ -84,6 +84,16 @@ def get_success_url(self: "PostCreateView") -> str:
         return reverse("post-detail", kwargs={"pk": self.object.pk})
 
 
+class PostDeleteView(DeleteView):
+    model = Post
+    template_name = "core/post-confirm-delete.html"
+    success_url = reverse_lazy("post-list")
+
+    def get_queryset(self: "PostDeleteView") -> models.QuerySet:
+        queryset = super().get_queryset()
+        return queryset.filter(author=self.request.user)
```

Comment:
DeleteView is using single objects, this method is not doing anything in my opinion

Reply:
This method is indeed important as it customizes the queryset to ensure that users can only delete posts they authored. This security measure prevents unauthorized deletions, maintaining the integrity of user data.